Public/Src/App/UnitTests/Bxl/ServerDeploymentTests.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Reflection;
+using BuildXL;
+using BuildXL.Engine;
+using BuildXL.Utilities;
+using Test.BuildXL.TestUtilities.Xunit;
+using Xunit;
+
+namespace Test.Bxl
+{
+    public class ServerDeploymentTests : TemporaryStorageTestBase
+    {
+        [Fact]
+        public void TestMissingManifestDirectory()
+        {
+            XAssert.IsTrue(ServerDeployment.IsServerDeploymentOutOfSync(TemporaryDirectory, null, out var deploymentDir));
+        }
+
+        [Fact]
+        public void TestMissingManifestFile()
+        {
+            var manifestRootDir = TemporaryDirectory;
+            var manifestPath = Path.Combine(manifestRootDir, AppDeployment.DeploymentManifestFileName);
+
+            File.WriteAllText(manifestPath, AssemblyHelper.GetAssemblyLocation(Assembly.GetExecutingAssembly()));
+
+            var appDeployment = AppDeployment.ReadDeploymentManifest(
+                Path.GetDirectoryName(manifestPath),
+                AppDeployment.DeploymentManifestFileName,
+                skipManifestCheckTestHook: true);
+
+            string deploymentDir = ServerDeployment.ComputeDeploymentDir(TemporaryDirectory);
+            Directory.CreateDirectory(deploymentDir);
+
+            XAssert.IsTrue(ServerDeployment.IsServerDeploymentOutOfSync(TemporaryDirectory, appDeployment, out deploymentDir));

```

Comment:
When an `out` variable is not used, it makes sense to say `out _` instead of using a named variable
```suggestion
            XAssert.IsTrue(ServerDeployment.IsServerDeploymentOutOfSync(TemporaryDirectory, appDeployment, out _));
```